astrosat_users/adapters.py

```diff
@@ -141,7 +141,11 @@ def get_email_confirmation_url(self, request, emailconfirmation):
         else:
             path = reverse("account_confirm_email", args=[emailconfirmation.key])
 
-        url = urljoin(request.META['HTTP_ORIGIN'], path)
+        if self.is_api and "HTTP_ORIGIN" in request.META:
+            # request originates from a client on a different domain...
+            url = urljoin(request.META['HTTP_ORIGIN'], path)
+        else:
+            url = build_absolute_uri(request, path)
```

Comment:
I guess this is because requests can come from the `API` app itself, i.e. `swagger`, is that why this check is here? Would it not work anyway without though, I would expect the `HTTP_ORIGIN` even to exist on those requests, but, if wrong, this makes perfect sense.

Reply:
`HTTP_ORIGIN` didn't exist on the test client.  I haven't checked in swagger; I will now.

Reply:
It _does_ work in swagger.